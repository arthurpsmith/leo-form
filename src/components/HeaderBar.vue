<template>
  <div>
    <v-app-bar
      app
      flat
      color="white"
    >
      <a href="/">
        <v-img src="https://assets.crossref.org/logo/crossref-logo-100.png" alt="Crossref logo" aspect-ratio="1.538"
             max-height="50px" max-width="72px" min-height="36px" min-width="72px"/>
      </a>
      <v-spacer/>
      <login-button/>
      <language-menu/>
    </v-app-bar>
  </div>

</template>

<script lang="ts">
import { defineComponent } from '@vue/composition-api'
import LanguageMenu from '@/components/LanguageMenu.vue'
import LoginButton from '@/components/LoginButton.vue'

export default defineComponent({
  name: 'HeaderBar',
  components: { LoginButton, LanguageMenu }
})
</script>

<style scoped>

</style>
