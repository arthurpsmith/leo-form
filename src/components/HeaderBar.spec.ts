import { createLocalVue, mount } from '@vue/test-utils'
import HeaderBar from '@/components/HeaderBar.vue'
import Vuetify from 'vuetify'
import CompositionApi from '@vue/composition-api'
import { authSymbol, getAuthService } from '@/statemachines/auth.machine'
import { useActor } from 'xstate-vue2'

describe('HeaderBar.vue', () => {
  // DO NOT use Vuetify on the localInstance
  // This is bootstrapped in the jest setup
  // file located in ./tests/setup.ts
  //
  // localVue.use(Vuetify)

  const localVue = createLocalVue()
  localVue.use(CompositionApi)
  let vuetify: Vuetify
  let authServiceToProvide: ReturnType<typeof getAuthService>
  let provideWithExportedSymbol: Record<string, unknown>
  let authService

  beforeEach(() => {
    vuetify = new Vuetify()
    authServiceToProvide = getAuthService()
    provideWithExportedSymbol = {
      [authSymbol as symbol]: authServiceToProvide
    }
    authService = useActor(authServiceToProvide)
  })

  it('should have a custom title and match snapshot', () => {
    const wrapper = mount(HeaderBar, {
      localVue,
      vuetify,
      provide: provideWithExportedSymbol,
      propsData: { title: 'Foobar' }
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()
  })
})
