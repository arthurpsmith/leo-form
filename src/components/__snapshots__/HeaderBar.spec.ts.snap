// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar.vue should have a custom title and match snapshot 1`] = `
<div title="Foobar">
  <header class="v-sheet theme--light v-toolbar v-toolbar--flat v-app-bar v-app-bar--fixed white" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 64px;"><a href="/">
        <div>
          <div aria-label="Crossref logo" role="img" class="v-image v-responsive theme--light" style="min-height: 36px; min-width: 72px; max-height: 50px; max-width: 72px;">
            <div class="v-responsive__sizer" style="padding-bottom: 65.01950585175553%;"></div>
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </a>
      <div class="spacer"></div>
      <div class="v-menu"><button type="button" class="v-btn v-btn--outlined theme--light v-size--default primary--text" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-translate theme--light"></i>
      $vuetify.language_menu_button
    </span></button>
        <!---->
      </div> <span><button type="button" class="ml-4 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">Log in</span></button>
      <!----></span>
    </div>
  </header>
</div>
`;
